### PR TITLE
Update README to clarify yarn build usage

### PR DESCRIPTION
**Pull request by axfelix**

### CODE REVIEW

**Review by flossypurse — APPROVED** [review]
🥳

File: README.md [hunk 1]
```diff
@@ -157,9 +157,10 @@ Format all files.
 yarn format
 ```
 
-Make sure you have a working build:
+Optionally, make sure you have a working build (the Github CI tools will check this too):
```
[the remark this answers is not in the document]
Comment (axfelix) [reply README.md:160]: Isn't yarn format redundant with yarn assemble right now too? Looks like assemble runs dprint now. Are we still trying to get https://github.com/temporalio/documentation/pull/2456 merged?

[the remark this answers is not in the document]
Comment (flossypurse) [reply README.md:160]: Well, we can have `yarn format` be a stand alone cmd to just format if you want.
But yes, `yarn assembly` now includes formatting.

There are some things from 2456 we might want - but some of it - I am not sure we do.

[the remark this answers is not in the document]
Comment (axfelix) [reply README.md:160]: that makes sense — but I like removing the dependency on `dprint` either way, especially if we're going to encourage contributors to use snipsync more in this repo. so let's figure out how much of that we want to keep before merging this

**Review by flossypurse — APPROVED** [review]

**Comment by axfelix** [comment]
@flossypurse see most recent commit -- are we good removing the `yarn format` step from every case where it previously followed `yarn assemble`? afaict that's redundant now. Also, not sure where those TOC changes came from in that most recent commit... got pulled in from merging main somewhere.

**Pull request merged**